app/controllers/samples_controller.rb

```diff
@@ -1265,4 +1268,13 @@ def sort_by(samples, dir = nil)
     end
     samples
   end
+
+  def check_owner
```

Comment:
this is here instead of ApplicationController b/c it's samples-specific and to access @sample